### DISCUSSION TITLE
Competition

### DISCUSSION THREAD

**Discussion by swyxio** [breadthe/sd-buddy#18] [General]
this may or may not be a goal of ours but people are coming close with their own GUIs:

https://news.ycombinator.com/item?id=32805396 (good list of features, some of which we have)

i'm wondering if we want to get to some minimum viable featureset and announce a 1.0

**Comment by breadthe** [comment]
Realistically I don't think I'll end up supporting all the stuff on the [wishlist](https://github.com/breadthe/sd-buddy#wishlist) but here are some things I definitely want for a 1.0.

* the parametric prompts we prototyped.
* img2txt.
* let's figure out what parameter combo can produce the same image each time - this one's been bugging me from the start.
* a UI wrapper with a few menus/sections, including Settings.
* ability to support/configure different environments? This one's a big if, since I don't have a good way to simulate what others are using.

I noticed a few of these apps floating around but they include Stable Diffusion along with the UI. It's interesting that they're going after the non-technical crowd with the "1-click" installer. It's not what I'm aiming for here, but it's good to keep in mind.

**Reply by swyxio** [reply]
- img2txt? you mean img2img? i dont think SD supports img2txt
- not sure i get the UI wrapper thing, but will just wait to see your vision

do let me know what you think i can contribute (do you think the parametric prompt stuff is ready? whats missing?), i think i might have the appetite to get more involved

**Reply by swyxio** [reply]
<details>
	<summary>
		ok i just ran this twice and got the same result:
	</summary>


![grid-0048](https://user-images.githubusercontent.com/6764957/189699395-43602eb4-bf71-4f16-bf2b-7c71d7346779.png)


</details>

`python scripts/txt2img.py --prompt "Anubis riding a motorbike in Grand Theft Auto cover, palm trees, cover art by Stephen Bliss, artstation, high quality" --ddim_steps=50 --n_samples=1 --n_rows=1 --n_iter=1 --seed 1805504473 --fixed_code`

can you try?

**Reply by breadthe** [reply]
Duh I meant `img2img`, though img2txt may be an idea for a new ML algo 😆 

For the UI, I want something like I did for [SVGX](https://svgx.app/). Top entry in the left sidebar would be txt2img, followed by img2img. Below that Help and Settings.

An alternative UI is to have tabs across the "header" area. For desktop/web apps I prefer the menu on the side because you typically have more real estate horizontally than vertically.

I haven't looked at img2img yet but if it shares a lot of params with txt2img it probably makes sense to combine them.

![grid-0414](https://user-images.githubusercontent.com/17433578/189739756-8bcc25e7-cfd4-465a-be9b-76d20de65dd3.png)

**Reply by swyxio** [reply]
ears a bit different and the buildings and the engines a bit different

**Comment by swyxio** [comment]
another https://twitter.com/_akhaliq/status/1570069426216603653/photo/1

**Comment by swyxio** [comment]
another https://twitter.com/charliebholtz/status/1571138577744138240

**Reply by breadthe** [reply]
Interesting. So these people package everything in one app. Their approach is much more accessible and doesn't require technical knowledge, at the expense of maintainability and huge installation size. My approach requires technical knowledge, but the UI itself is 4MB. Eh, I made it mostly for people like me so I'm ok with that I guess.
